src/lemlib/chassis/motions/moveToPoint.cpp

```diff
@@ -6,126 +6,135 @@
+            lateralLargeExit.update(lateralError);
+
+            // get output from PIDs
+            float lateralOut = lateralPID.update(lateralError);
+            float angularOut = angularPID.update(radToDeg(angularError));
+            if (close) angularOut = 0;
+
+            // apply restrictions on angular speed
+            angularOut = std::clamp(angularOut, -params.maxSpeed, params.maxSpeed);
+            angularOut = slew(angularOut, prevAngularOut, angularSettings.slew);
+
+            // apply restrictions on lateral speed
+            lateralOut = std::clamp(lateralOut, -params.maxSpeed, params.maxSpeed);
+            // constrain lateral output by max accel
+            // but not for decelerating, since that would interfere with settling
+            if (!close) lateralOut = slew(lateralOut, prevLateralOut, lateralSettings.slew);
+
+            // prevent moving in the wrong direction
+            if (params.forwards && !close) lateralOut = std::fmax(lateralOut, 0);
+            else if (!params.forwards && !close) lateralOut = std::fmin(lateralOut, 0);
+
+            // constrain lateral output by the minimum speed
+            if (params.forwards && lateralOut < fabs(params.minSpeed) && lateralOut > 0)
+                lateralOut = fabs(params.minSpeed);
+            if (!params.forwards && -lateralOut < fabs(params.minSpeed) && lateralOut < 0)
+                lateralOut = -fabs(params.minSpeed);
+
+            // update previous output
+            prevAngularOut = angularOut;
+            prevLateralOut = lateralOut;
+
+            infoSink()->debug("Angular Out: {}, Lateral Out: {}", angularOut, lateralOut);
+
+            // ratio the speeds to respect the max speed
+            float leftPower = lateralOut + angularOut;
+            float rightPower = lateralOut - angularOut;
+            const float ratio = std::max(std::fabs(leftPower), std::fabs(rightPower)) / params.maxSpeed;
+            if (ratio > 1) {
+                leftPower /= ratio;
+                rightPower /= ratio;
+            }
+
+            // move the drivetrain
+            drivetrain.leftMotors->move(leftPower);
+            drivetrain.rightMotors->move(rightPower);
+
+            // delay to save resources
+            pros::delay(10);
         }
 
-        // move the drivetrain
-        drivetrain.leftMotors->move(leftPower);
-        drivetrain.rightMotors->move(rightPower);
-
-        // delay to save resources
-        pros::delay(10);
+        // stop the drivetrain
+        drivetrain.leftMotors->move(0);
+        drivetrain.rightMotors->move(0);
+        // set distTraveled to -1 to indicate that the function has finished
+        distTraveled = -1;
+        this->endMotion();
+    });
+    pros::delay(10);
+    startedMutex.take(TIMEOUT_MAX);
+    if (!async) {
+        do pros::delay(10);
+        while (!(task.get_state() == pros::E_TASK_STATE_INVALID || task.get_state() == pros::E_TASK_STATE_SUSPENDED));
```

Comment:
This should check if the task was deleted too
```suggestion
        while (!(task.get_state() == pros::E_TASK_STATE_INVALID || task.get_state() == pros::E_TASK_STATE_SUSPENDED || task.get_state() == pros::E_TASK_STATE_DELETED));
```